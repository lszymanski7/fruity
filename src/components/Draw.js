import React from 'react'
import PropTypes from 'prop-types'
import { SHAPES } from '../data/constants'

const Draw = ({ animation, disabled, handleDraw, handleReset }) => {
    return (
        <div className="flexbox-row draw">
            <button
                className="draw__start-button"
                disabled={disabled}
                onClick={handleDraw}
                type="button"
            >
                {disabled ? 'SELECT MORE FRUITS' : 'DRAW A FRUIT'}
            </button>
            <button
                aria-label="Reset"
                className={'draw__reset-button' + (animation ? ' ' + animation : '')}
                disabled={disabled}
                onClick={handleReset}
                type="button"
            >
                <img
                    alt="Reset Icon"
                    className="draw__reset-icon"
                    src={SHAPES.reset}
                />
            </button>
        </div>
    )
}

Draw.propTypes = {
    animation: PropTypes.string,
    disabled: PropTypes.bool.isRequired,
    handleDraw: PropTypes.func.isRequired,
    handleReset: PropTypes.func.isRequired
}

export { Draw as default }
