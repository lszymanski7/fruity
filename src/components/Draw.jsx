import React from 'react'
import PropTypes from 'prop-types'
import { SHAPES } from '../data/constants'

const Draw = ({ animation, disabled, handleDraw, handleReset }) => {
  return (
    <div className="draw">
      <div className="draw__buttons-container">
        <button
          className="draw__start-button"
          disabled={disabled}
          onClick={handleDraw}
          type="button"
        >
          {disabled ? 'SELECT MORE FRUITS' : 'DRAW A FRUIT'}
        </button>
        <button
          aria-label="Reset"
          className={'draw__reset-button' + (animation ? ' ' + animation : '')}
          disabled={animation !== null}
          onClick={handleReset}
          type="button"
        >
          <img
            alt="Reset Icon"
            className="draw__reset-icon"
            src={SHAPES.reset}
          />
        </button>
      </div>
    </div>
  )
}

Draw.propTypes = {
  animation: PropTypes.string,
  disabled: PropTypes.bool.isRequired,
  handleDraw: PropTypes.func.isRequired,
  handleReset: PropTypes.func.isRequired
}

export default Draw
