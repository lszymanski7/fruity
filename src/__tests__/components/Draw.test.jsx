import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Draw from '../../components/Draw'

const props = {
  animation: null,
  disabled: false,
  handleDraw: jest.fn(),
  handleReset: jest.fn()
}

describe('Draw', () => {
  it('should be rendered correctly.', () => {
    const { container } = render(<Draw {...props} />)
    expect(container).toMatchSnapshot()
  })

  describe('Button (draw)', () => {
    it('should handle clicking.', async () => {
      render(<Draw {...props} />)
      const button = screen.getAllByRole('button')[0]
      await userEvent.click(button)
      expect(props.handleDraw).toHaveBeenCalled()
    })

    it('should have the correct text when enabled.', () => {
      render(<Draw {...props} />)
      const button = screen.getAllByRole('button')[0]
      expect(button).toBeEnabled()
      expect(button).toHaveTextContent('DRAW A FRUIT')
    })

    it('should have the correct text when disabled.', () => {
      render(<Draw {...props} disabled={true} />)
      const button = screen.getAllByRole('button')[0]
      expect(button).toBeDisabled()
      expect(button).toHaveTextContent('SELECT MORE FRUITS')
    })
  })

  describe('Button (reset)', () => {
    it('should handle clicking.', async () => {
      render(<Draw {...props} />)
      const button = screen.getAllByRole('button')[1]
      await userEvent.click(button)
      expect(props.handleReset).toHaveBeenCalled()
    })

    it('should not have an animation when enabled.', () => {
      render(<Draw {...props} />)
      const button = screen.getAllByRole('button')[1]
      expect(button).toBeEnabled()
      expect(button).not.toHaveClass('spin')
    })

    it('should have an animation when disabled.', () => {
      render(<Draw {...props} animation={'spin'} />)
      const button = screen.getAllByRole('button')[1]
      expect(button).toBeDisabled()
      expect(button).toHaveClass('spin')
    })

    it('should have the correct background image.', () => {
      render(<Draw {...props} />)
      const img = screen.getByRole('img')
      expect(img).toHaveAttribute('alt', 'Reset Icon')
      expect(img).toHaveAttribute('src', 'reset.svg')
    })
  })
})
