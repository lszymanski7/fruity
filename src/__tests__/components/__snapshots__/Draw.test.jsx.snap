// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draw should be rendered correctly. 1`] = `
<div>
  <div
    class="draw"
  >
    <div
      class="draw__buttons-container"
    >
      <button
        class="draw__start-button"
        type="button"
      >
        DRAW A FRUIT
      </button>
      <button
        aria-label="Reset"
        class="draw__reset-button"
        type="button"
      >
        <img
          alt="Reset Icon"
          class="draw__reset-icon"
          src="reset.svg"
        />
      </button>
    </div>
  </div>
</div>
`;
